sphinx_needs/needs.py

```diff
@@ -775,28 +781,34 @@ def check_configuration(app: Sphinx, config: Config) -> None:
             )
 
     external_variants = needs_config.variants
-    external_variant_options = needs_config.variant_options
     for value in external_variants.values():
         # Check if external filter values is really a string
         if not isinstance(value, str):
             raise NeedsConfigException(
                 f"Variant filter value: {value} from needs_variants {external_variants} is not a string."
             )
 
-    allowed_internal_variants = {
-        k for k, v in NeedsCoreFields.items() if v.get("allow_variants")
-    }
-    for option in external_variant_options:
-        # Check variant option is added to an allowed field
-        if option in link_types:
-            raise NeedsConfigException(
-                f"Variant option `{option}` is a link type. This is not allowed."
-            )
-        if option not in extra_options and option not in allowed_internal_variants:
-            raise NeedsConfigException(
-                f"Variant option `{option}` is not added in extra options. "
-                "This is not allowed."
-            )
+    if needs_config._variant_options:
+        log_warning(
+            LOGGER,
+            'Config option "needs_variant_options" is deprecated. Please use "needs_fields" with "parse_variants" instead.',
+            "deprecated",
+            None,
+        )
+        allowed_internal_variants = {
+            k for k, v in NeedsCoreFields.items() if v.get("allow_variants")
+        }
+        for option in needs_config._variant_options:
+            # Check variant option is added to an allowed field
+            if option in link_types:
+                raise NeedsConfigException(
+                    f"Variant option `{option}` is a link type. This is not allowed."
+                )
+            if option not in extra_options and option not in allowed_internal_variants:
+                raise NeedsConfigException(
+                    f"Variant option `{option}` is not added in extra options. "
```

Comment:
Line-Suggestions stop working for me... we should rephrase this to point to `needs_fields`.